### PR TITLE
Update tanstack dependencies

### LINKED ISSUE #1617: Incoming dependency update breaks table row rendering
## 🧹 Tech Debt

I tried updating several dependencies to latest in [this SystemLinkShared PR](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/582622) and [this Nimble PR](https://github.com/ni/nimble/pull/1615). In some cases there are unit test failures and Chromatic diffs caused by tables that no longer render any rows.

It appears not to break every use case:
- in the SystemLinkShared pipeline there are 15 unit test failures in SlTableComponent but the majority of unit tests still pass. It didn't run Chromatic but I saw locally that the SlTable story doesn't render rows while the data table story does 🤷.
-  in the Nimble pipeline it's only showing Chromatic diffs for the label provider stories, not the table stories. And no unit tests failed.

The update includes these bumps:
- `@tanstack/table-core`  8.10.3 -> 8.10.7 
- `@tanstack/virtual-core`  3.0.0-beta.61 -> 3.0.0-beta.68
- changes to d3 and tiptap which are likely unrelated

### PR DESCRIPTION
**Pull request by mollykreis**
# Pull Request

## 🤨 Rationale

An issue was discovered when trying to automatically update the dependency on TanStack. This PR updates those dependencies and fixes the issue on the label-provider storybook page.

Resolves #1617

## 👩‍💻 Implementation

- Update version of `@tanstack/table-core` and `@tanstack/virtual-core`
- Set a `min-height` on the tables used in the label-provider stories

## 🧪 Testing

Verified the tests pass and there are no chromatic changes

## ✅ Checklist

- [ ] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Comment by mollykreis** [comment]
@atmgrifter00, can you buddy this for me?

File: packages/nimble-components/src/label-provider/base/tests/label-provider-stories-utils.ts [hunk 1]
```diff
@@ -63,6 +63,7 @@ export const labelProviderMetadata: Meta<LabelProviderArgs> = {
                 }
                 ${tableTag} {
                     height: auto;
+                    min-height: 100px;
```
Comment (atmgrifter00) [inline packages/nimble-components/src/label-provider/base/tests/label-provider-stories-utils.ts:66]: Would it be worth leaving a comment as to why this is necessary?

**Review by atmgrifter00 — APPROVED** [review]

File: change/@ni-nimble-components-54d4e9df-8176-49b5-9021-269b73561000.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (jattasNI) [inline change/@ni-nimble-components-54d4e9df-8176-49b5-9021-269b73561000.json:1]: Is there a new (or newly discovered) expectation on how clients set the table's height? If so should we try to document it in the table's story?

Something like:
## Sizing
Clients must provide an explicit `height` or `min-height` for `nimble-table`. Do not leave both values set to `auto`.

**Review by jattasNI — APPROVED** [review]

**Comment by mollykreis** [comment]
@atmgrifter00, do you mind taking another buddy pass of this PR? I've changed the approach.

**Review by atmgrifter00 — APPROVED** [review]

File: packages/nimble-components/src/table/tests/table.mdx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+<Description of={tableStories} />
+
+<DocsStory of={tableStories.table} expanded={false} />
+<Controls of={tableStories.table} />
+
+# Usage Docs
+
+## Styling
+
+The `nimble-table` does not currently support being styled with `height: auto`. The ability to auto size the table is tracked with [issue 1624](https://github.com/ni/nimble/issues/1624).
```
Comment (rajsite) [inline packages/nimble-components/src/table/tests/table.mdx:23]: I'd start with the suggestion and follow with exceptions
```suggestion
The `nimble-table` should be sized explicitly or sized to fill the space of a parent container. The `nimble-table` does not currently support being styled with `height: auto`. The ability to auto size the table is tracked with [issue 1624](https://github.com/ni/nimble/issues/1624).
```

**Review by rajsite — APPROVED** [review]

**Review by jattasNI — APPROVED** [review]

**Pull request merged**